frontend/src/v5/ui/routes/viewer/tickets/ticketsList/ticketsList.component.tsx

```diff
@@ -49,13 +55,23 @@ export const TicketsList = ({ tickets }: TicketsListProps) => {
 		setSelectedTemplates(new Set(selectedTemplates));
 	};
 
-	const getTicketsByTemplateId = (templateId: string) => tickets.filter(({ type }) => type === templateId);
+	const filterCompleted = (ticket) => {
+		const issuePropertyStatus = get(ticket, 'properties.Status');
+		const treatmentStatus = get(ticket, 'modules.safetibase.Treatment Status');
+
+		const isCompletedIssueProperty = [TicketStatuses.CLOSED, TicketStatuses.VOID].includes(issuePropertyStatus);
+		const isCompletedTreatmentStatus = [TreatmentStatuses.AGREED_FULLY, TreatmentStatuses.VOID].includes(treatmentStatus);
 
-	const getFilteredTickets = () => {
-		if (selectedTemplates.size === 0) return tickets;
-		return flatMap([...selectedTemplates], getTicketsByTemplateId);
+		return (isCompletedIssueProperty || isCompletedTreatmentStatus) === showingCompleted;
 	};
 
+	const getTicketsByTemplateId = (templateId: string) => tickets.filter(({ type }) => type === templateId);
+
+	const getFilteredTickets = (() => {
+		if (selectedTemplates.size === 0) return tickets.filter(filterCompleted);
+		return flatMap([...selectedTemplates], getTicketsByTemplateId).filter(filterCompleted);
```

Comment:
This would get rid of the getting filtered mutiple times.
```suggestion
tickets.filter((ticket) => (!selectedTemplates.size || selectedTemplates.has(ticket.type)) && filterCompleted(ticket))
```